lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -144,8 +146,12 @@ def index(self, trans, deleted='False', **kwd):
         filters = []
         # support the old default of not-returning/filtering-out deleted histories
         filters += self._get_deleted_filter(deleted, filter_params)
-        # users are limited to requesting only their own histories (here)
-        filters += [self.app.model.History.user == current_user]
+        # check if user is admin and optional parameter 'all' is True
+        admin = trans.user_is_admin
+        all_histories = util.string_as_bool(kwd.get('all', False))
```

Comment:
Can you throw ``galaxy.exceptions.AdminRequiredException`` if all_histories is true but admin is not? I think the new if would read a bit better too if you did that.

Reply:
Added. Thanks for the exception tip.